packages/cli/src/runners/task-broker.service.ts

```diff
@@ -408,18 +416,45 @@ export class TaskBroker {
 
 	async sendTaskSettings(taskId: Task['id'], settings: unknown) {
 		const runner = await this.getRunnerOrFailTask(taskId);
+
+		const task = this.tasks.get(taskId);
+
+		if (!task) return;
+
+		task.timeout = setTimeout(async () => {
+			await this.handleTaskTimeout(taskId);
+		}, this.config.taskTimeout * Time.seconds.toMilliseconds);
+
 		await this.messageRunner(runner.id, {
 			type: 'broker:tasksettings',
 			taskId,
 			settings,
 		});
 	}
 
+	private async handleTaskTimeout(taskId: Task['id']) {
+		const task = this.tasks.get(taskId);
+
+		if (!task) return;
+
+		clearTimeout(task.timeout);
+
+		this.runnerLifecycleEvents.emit('runner:timed-out-during-task');
+
+		const timeoutError = new ApplicationError(
```

Comment:
Now this error is:

<img width="474" alt="image" src="https://github.com/user-attachments/assets/74b42833-10d0-4dbf-80ff-a0aad78b16ff">

Could we have some more user friendly message with some suggestions, similar to `TaskRunnerOomError`?